### DISCUSSION TITLE
Will turning of certain object detections for a camera have a positive impact on CPU ?

### DISCUSSION THREAD

**Discussion by atv2016** [blakeblackshear/frigate#5763] [General]
Was just wondering where i can optimise, or if it's not worth removing these things (a car in the backyard, etc)

**Comment by NickM-27** [comment]
The image is scanned for all objects so it won't inherently increase usage but if it did detect a car in the backyard as a false positive it would continue to check for that object which would use more resources. I also just think it's better for organization sake to keep object lists specific.

**Reply by atv2016** [reply]
Makes sense, thanks

**Reply by ediguidedog** [reply]
> The image is scanned for all objects so it won't inherently increase usage but if it did detect a car in the backyard as a false positive it would continue to check for that object which would use more resources. I also just think it's better for organization sake to keep object lists specific.

Just so I'm clear.  Is the image passed to the coral or whatever with a list of objects of interest and a match is then found or not found with the result returned? Or is the image passed to the coral and it returns any possible matches from the model and then frigate decides if the detected objects are of use?

**Reply by NickM-27** [reply]
> > The image is scanned for all objects so it won't inherently increase usage but if it did detect a car in the backyard as a false positive it would continue to check for that object which would use more resources. I also just think it's better for organization sake to keep object lists specific.
> 
> Just so I'm clear. Is the image passed to the coral or whatever with a list of objects of interest and a match is then found or not found with the result returned? Or is the image passed to the coral and it returns any possible matches from the model and then frigate decides if the detected objects are of use?

It is the latter, after the model is loaded the only thing passed to the coral is the image and then the coral returns all labels and then frigate filters through based on labels that are included in the config.

**Reply by ediguidedog** [reply]
Thanks.
